.github/workflows/update-base-image.yml

```diff
@@ -0,0 +1,53 @@
+name: Update Base Image
+on:
+  schedule:
+    - cron: 0 1 * * 1
+  workflow_dispatch:
+jobs:
+  update-base-image:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Update Base Image
```

Comment:
may want to extend this to do java too - don't have to do it in this PR but maybe rename job/workflow to make them a bit more generic?